### PR TITLE
llvm 18 build fixes.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Comment by devnexen** [comment]
@vanhauser-thc gonna love it :) I can kind of see why they removed the get*PtrTy shortcuts, but the starts_with/startswith situation is harder to comprehend .. we ll see.

**Comment by vanhauser-thc** [comment]
oh my what a shit show ... so many modules that need to be changed ...

**Comment by devnexen** [comment]
I think I m done, with the last version from apt.llvm the build and tests pass locally.

File: instrumentation/afl-llvm-dict2file.so.cc [hunk 1]
```diff
@@ -433,15 +433,15 @@ bool AFLdict2filePass::runOnModule(Module &M) {
...
           isStrcmp &=
               FT->getNumParams() == 2 && FT->getReturnType()->isIntegerTy(32) &&
               FT->getParamType(0) == FT->getParamType(1) &&
-              FT->getParamType(0) == IntegerType::getInt8PtrTy(M.getContext());
+              FT->getParamType(0) == IntegerType::getInt8Ty(M.getContext())->getPointerTo(0);
           isStrcasecmp &=
               FT->getNumParams() == 2 && FT->getReturnType()->isIntegerTy(32) &&
               FT->getParamType(0) == FT->getParamType(1) &&
-              FT->getParamType(0) == IntegerType::getInt8PtrTy(M.getContext());
+              FT->getParamType(0) == IntegerType::getInt8Ty(M.getContext())->getPointerTo(0);
```
Comment (vanhauser-thc) [inline instrumentation/afl-llvm-dict2file.so.cc:444]: Does this not need an ifdef? Or does this work with llvm 11/13 too?

Comment (devnexen) [reply instrumentation/afl-llvm-dict2file.so.cc:444]: Not at all, what the get*PtrTy did is precisely this, they were just shortcuts. llvm 18 feels like "api cleanup" (getFixedSize which calls geFixedValue is the same vein).

**Comment by vanhauser-thc** [comment]
Thank you!

**Pull request merged**